src/ngraph/serializer.cpp

```diff
@@ -492,1228 +526,1246 @@ T get_value(nlohmann::json js, const string& key)
     return rc;
 }
 
-static shared_ptr<ngraph::Function>
-    read_function(const json& func_js,
-                  unordered_map<string, shared_ptr<Function>>& function_map,
-                  function<const_data_callback_t> const_data_callback)
+shared_ptr<Node> JSONDeserializer::deserialize_node_reference(json& j)
 {
-    shared_ptr<ngraph::Function> rc;
+    const string& name = j;
+    return m_node_map.at(name);
+}
 
+shared_ptr<Function> JSONDeserializer::deserialize_function(json& func_js)
+{
     string func_name = func_js.at("name").get<string>();
-    vector<string> func_parameters = func_js.at("parameters").get<vector<string>>();
-    vector<string> func_result = func_js.at("result").get<vector<string>>();
-    unordered_map<string, shared_ptr<Node>> node_map;
+    vector<json> func_parameters = func_js.at("parameters"); //.get<vector<json>>();
```

Comment:
Does get<vector<json>>() not work if you include it. Can you just include that code uncommented? If not then delete the comment.

Reply:
Those work; I forgot to delete them. Will do.